From 0d39e71f981f413e952262cae4c4b41b60a23308 Mon Sep 17 00:00:00 2001
From: Kamil Jarosz <kjarosh256@gmail.com>
Date: Thu, 10 Sep 2026 22:44:43 +0200
Subject: [PATCH] desktop: Fix 32-bit builds with freetype

FreeType types differ between architectures, so take that into account.
---
 frontend-utils/src/backends/ui/freetype_font_renderer.rs | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/frontend-utils/src/backends/ui/freetype_font_renderer.rs b/frontend-utils/src/backends/ui/freetype_font_renderer.rs
index 270400ec6163..990b531fca3d 100644
--- a/frontend-utils/src/backends/ui/freetype_font_renderer.rs
+++ b/frontend-utils/src/backends/ui/freetype_font_renderer.rs
@@ -169,7 +169,11 @@ impl FontRenderer for FreetypeFontRenderer {
 
 /// Converts a FreeType 26.6 fixed-point value (1/64th of a pixel) to Twips
 /// (1/20th of a pixel), rounding to the nearest twip.
-fn convert_26_6_to_twips(value_26_6: i64) -> Twips {
+///
+/// Note: the FreeType type differs between 64-bit and 32-bit architectures,
+/// hence the `Into<i64>`.
+fn convert_26_6_to_twips(value_26_6: impl Into<i64>) -> Twips {
+    let value_26_6 = value_26_6.into();
     let whole_pixels = (value_26_6 / 64) as i32;
 
     let fractional_twips = (value_26_6 % 64) as f64 / 64.0 * 20.0;
